packages/spacecat-shared-gpt-client/src/clients/firefall-client.js

```diff
@@ -161,23 +163,113 @@ export default class FirefallClient {
     return jobStatusResponse;
   }
 
-  async fetch(prompt) {
+  /**
+   * Fetches data from Firefall Chat Completion API.
+   * @param prompt The text prompt to provide to Firefall
+   * @param options The options for the call:
+   *                - imageUrls: An array of URLs of the images to provide to Firefall
+   * @returns {Promise<*>}
+   */
+  async fetchChatCompletion(prompt, options = {}) {
     if (!hasText(prompt)) {
       throw new Error('Invalid prompt received');
     }
+    const hasImageUrls = options?.imageUrls && options.imageUrls.length > 0;
 
     try {
       const startTime = process.hrtime.bigint();
-      const jobSubmissionResponse = await this.#submitJob(prompt);
-      const jobStatusResponse = await this.#pollJobStatus(jobSubmissionResponse.job_id);
-      this.#logDuration('Firefall API call', startTime);
+      const userRole = {
+        role: 'user',
+        content: [
+          {
+            type: 'text',
+            text: prompt,
+          },
+        ],
+      };
+      const body = {
+        llm_metadata: {
+          model_name: this.config.FIREFALL_API_CAPABILITY_NAME,
+          llm_type: 'azure_chat_openai',
+        },
+        messages: [
+          {
+            role: 'system',
+            content: 'You are a helpful assistant designed to output JSON.',
```

Comment:
This system role is only need when a response format of `json_object` is requested.

Reply:
I assumed it would always return a JSON object.  I believe that is the only requirement right now.  If others need another response, should we anticipate and allow it right now?  Or allow a future enhancement for it?

Reply:
This isn't the HTTP response format, it is the message format. We are already using both formats with the import assistant.